### PR TITLE
Change workspace integration test from core-dev to preview environment

### LINKED ISSUE #9749: Change workspace integration tests nightly from core-dev to preview environment
### Bug description

The platform team recently (May 2) change the default environment as preview environment, deprecated the core-dev.
https://gitpod.slack.com/archives/C032A46PWR0/p1651473669803169

Therefore, the nightly integration test can't run successfully.

### Steps to reproduce

Run the werft job
```shell
werft run github -j .werft/workspace-integration-tests-startup-cron.yaml
```

### Workspace affected

_No response_

### Expected behavior

_No response_

### Example repository

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by jenting**
## Description
Since the preview environment is now the default environment that the werft job deploy to.
Change the workspace integration test being able to run against preview environment.

The werft job is still running in the `core-dev` context `werft` namespace though.
I did not find a way to make it possible that werft job being able to run on another context unless we run the integration test job by manually applies the yaml manifest to `preview-environment` context `default` namespace within the werft job.
But it also means we have to copy the related [secrets](https://github.com/gitpod-io/gitpod/blob/137f5430090ea96564ee4f88232650526009cacc/.werft/workspace-run-integration-tests.yaml#L60-L79) from `core-dev` to `preview environment` context.

Therefore, I give up to find a simple way to make the integration test pod being able to run in `preview-environment` context. Instead, make the integration test being able to run on any machine against the `preview-environment` context `default` namespace. But it also mean that we need to fix these [two integration tests](https://github.com/gitpod-io/gitpod/blob/137f5430090ea96564ee4f88232650526009cacc/test/README.md?plain=1#L72) #9953 since previously design only allow the test be passed when the integration test pod running within the same cluster. (I believe the integration test should be passed regardless of the environment in which it is executed).

In general, it runs `go test ...` in `core-dev` context `werft` namespace against `preview-environment` context `default` namespace. It's like what we run `go test ..` locally against preview-environment` context `default` namespace.

## Related Issue(s)

Fixes #9749

## How to test

```shell
werft run github -j .werft/workspace-run-integration-tests.yaml
```

Reference werft job https://werft.gitpod-dev.com/job/gitpod-custom-jenting-change-workspace-nightly-9749.55
Reference slack notification https://gitpod.slack.com/archives/C03E52788SU/p1652399601400079

## Release Notes

```release-note
None
```

## Documentation

N/A

### CODE REVIEW

**Comment by princerachit** [comment]
@jenting can you describe the changes you made in this PR. I see you removed a file why so? What was the reason of getting rid of the args and other fields in the YAML?

**Comment by jenting** [comment]
> @jenting can you describe the changes you made in this PR. I see you removed a file why so? What was the reason of getting rid of the args and other fields in the YAML?

I've update the PR description, hope it clear to understand 🙏

**Comment by csweichel** [comment]
> I believe the integration test should be passed regardless of the environment in which it is executed

💯 🎯

**Comment by jenting** [comment]
/unhold

Ready to review

File: .werft/workspace-run-integration-tests.yaml [hunk 1]
```diff
@@ -77,59 +30,140 @@ pod:
...
+      BRANCH="inte-test/"$(date +%Y%m%d%H%M%S)
+      export WERFT_CREDENTIAL_HELPER=/workspace/dev/preview/werft-credential-helper.sh
 
-      echo "[prep] receiving config..."
-      export GOOGLE_APPLICATION_CREDENTIALS="/config/gcloud/legacy_credentials/cd-gitpod-deployer@gitpod-core-dev.iam.gserviceaccount.com/adc.json"
-      echo "[prep] received config."
+      function cleanup ()
+      {
+        werft log phase "clean up" "clean up"
+        git push origin :$BRANCH | werft log slice "clean up"
```
Comment (sagor999) [inline .werft/workspace-run-integration-tests.yaml:62]: nit: can you double quote all variable names to prevent word splitting

**Review by sagor999 — APPROVED** [review]
LGTM except for minor nit.

/hold

feel free to remove hold when\if addressed that nit.

File: .werft/workspace-run-integration-tests.yaml (hunk 1 above)
Comment (jenting) [reply .werft/workspace-run-integration-tests.yaml]: addressed

**Pull request merged**